client/landing/stepper/hooks/test/use-site-migration-key.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * @jest-environment jsdom
+ */
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import nock from 'nock';
+import React from 'react';
+import { useSiteMigrationKey } from '../use-site-migraiton-key';
+
+describe( 'useSiteMigrationKey', () => {
+	it( 'returns the site migration key', async () => {
+		const queryClient = new QueryClient();
+		const wrapper = ( { children } ) => (
+			<QueryClientProvider client={ queryClient }>{ children }</QueryClientProvider>
+		);
+
+		nock( 'https://public-api.wordpress.com' )
+			.get( '/wpcom/v2/sites/some-site-id/atomic-migration-status/migrate-guru-key' )
+			.query( { http_envelope: 1 } )
+			.once()
+			.reply( 200, { migration_key: 'some-migration-key' } );
+
+		const { result } = renderHook( () => useSiteMigrationKey( 'some-site-id' ), { wrapper } );
```

Comment:
Nit: the argument expects the siteId argument to be numeric, so we should likely test it with a number (or allow a site slug to be specified in the types).

Reply:
Fixed here https://github.com/Automattic/wp-calypso/pull/88952/commits/67c80a1df84213e1ab9cc8af355b9eba6d327ef9

We are with the typescript checking disabled on the test files on our pipeline😢 .